nautobot/extras/fields.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from nautobot.utilities.forms import DynamicModelChoiceField
+
+
+class ForeignKeyLimitedByContentTypes(models.ForeignKey):
```

Comment:
Trying to curtail the sprawl of additional items in `extras` (seeing as we can't accomplish #2722) in 2.0, can this live in `core`?